facebook-commerce.php

```diff
@@ -785,26 +790,26 @@ private function find_variation_product_item_ids( $product_group_id ) {
 		return $product_item_ids;
 	}
 
-
 	/**
-	 * Gets the total of published products.
+	 * Gets the total number of published products.
 	 *
 	 * @return int
 	 */
 	public function get_product_count() {
+		$product_counts = wp_count_posts( 'product' );
+		return $product_counts->publish;
+	}
 
-		$args = array(
-			'post_type'      => 'product',
-			'post_status'    => 'publish',
-			'posts_per_page' => -1,
-			'fields'         => 'ids',
-		);
-
-		$products = new WP_Query( $args );
-
-		wp_reset_postdata();
```

Comment:
I've reimplemented this function so it's more memory efficient. Previously the query would load all product IDs; `wp_count_posts()` uses a `COUNT()` SQL statement and doesn't need to load any actual data.

This is not essential to the success of this PR but seems a useful improvement while we're here!